presto-spi/src/main/java/com/facebook/presto/spi/type/Type.java

```diff
@@ -145,4 +145,8 @@
      * Compare the values in the specified block at the specified positions equal.
      */
     int compareTo(Block leftBlock, int leftPosition, Block rightBlock, int rightPosition);
+
+    Object getObject(Block block, int position);
```

Comment:
Add javadoc for these two methods, and I would move them to the end of the get* and write* sections, just to match the way the code is currently laid out